tests/phpunit/src/TestBase.php

```diff
@@ -85,37 +84,50 @@ abstract class TestBase extends TestCase {
 
   /** @var array */
   protected $cloudConfig = [];
+  /**
+   * @var \Symfony\Component\Console\Output\BufferedOutput|null
+   */
+  protected $applicationOutput;
+
+  /**
+   * @var string
+   */
+  protected $dataDir;
+
+  /**
+   * @var string
+   */
+  protected $cloudConfigFilepath;
+
+  /**
+   * @var string
+   */
+  protected $acliConfigFilepath;
 
   /**
    * This method is called before each test.
    *
    * @param null $output
    *
-   * @throws \Psr\Cache\InvalidArgumentException
+   * @throws \Exception
    */
   protected function setUp($output = NULL): void {
     if (!$output) {
       $output = new BufferedOutput();
     }
+    $this->applicationOutput = $output;
 
     $this->fs = new Filesystem();
     $this->prophet = new Prophet();
     $this->consoleOutput = new ConsoleOutput();
-    $this->input = new ArrayInput([]);
-    $logger = new ConsoleLogger($output);
     $this->fixtureDir = realpath(__DIR__ . '/../../fixtures');
     $this->projectFixtureDir = $this->fixtureDir . '/project';
+    $this->dataDir = $this->fixtureDir . '/.acquia';
     $this->amplitudeProphecy = $this->prophet->prophesize(Amplitude::class);
     $this->clientProphecy = $this->prophet->prophesize(Client::class);
 
-    $container = new ContainerBuilder();
-    $container->setParameter('repo_root', $this->projectFixtureDir);
-    $container->setParameter('data_dir', $this->fixtureDir . '/.acquia');
-    AcquiaCliApplication::configureContainer($container, $this->input, $output, $logger);
-    $this->configureContainer($container);
-
-    $this->application = new AcquiaCliApplication($container, $logger, $this->input, $output, 'UNKNOWN');
```

Comment:
We delay building the container and creating the application until just before command execution.